horace_core/sqw/@sqw/tobyfit_DGdisk_resconv.m

```diff
@@ -115,16 +117,63 @@
 refine_crystal = ~isempty(xtal);
 refine_moderator = ~isempty(modshape);
 if refine_crystal && refine_moderator
-    error('Cannot refine both crystal and moderator parameters. Error in logic flow - this should have been caught')
+    error('HORACE:tobyfit_resconv:invalid_argument', ...
+          'Cannot refine both crystal and moderator parameters. Error in logic flow - this should have been caught')
 end
 
-
 % Initialise output arguments
 % ---------------------------
 wout = copy(win);
 state_out = cell(size(win));    % create output argument
 store_out = [];
 
+% Perform resolution broadening calculation
+% -----------------------------------------
+% Package parameters as a cell for convenience
+if ~iscell(pars)
+    pars={pars};
+end
+
+reset_state=caller.reset_state;
+
+% Factor 10 because of 44 x npix array
+max_pix_size = get(hor_config, 'mem_chunk_size') / 10;
+
+for i=1:numel(ind)
+    % Get index of workspace into lookup tables
+    iw=ind(i);
+
+    % Set random number generator if necessary, and save if required for later
+    if reset_state
+        if ~isempty(state_in{i})
+            rng(state_in{i})
+        end
+    else
+        state_out{i} = rng;     % capture the random number generator state
+    end
+
+    npix = win(i).data.npix;
+    [npix_chunks, idxs] = split_vector_fixed_sum(npix, max_pix_size);
+    pix_bin_regions = [1, cumsum(cellfun(@sum, npix_chunks))];
+
+    for j = 1:numel(pix_bin_regions)-1
+        wout(i) = compute_resconv(wout(i), iw, lookup, sqwfunc, pars, ...
+                                  mc_contributions, mc_points, ...
+                                  refine_crystal, xtal, ...
+                                  refine_moderator, modshape, ...
+                                  pix_bin_regions(j), pix_bin_regions(j+1));
+
+    end
+    wout(i) = recompute_bin_data(wout(i));
+end
+
+end
+
+function sqw_obj = compute_resconv(sqw_obj, iw, lookup, sqwfunc, pars, ...
```

Comment:
I like to see a clear demarcation between one function and another in a file - typically two blank lines and then a line of 80 minus signs. It helps to make the code readable.